test/resource.cpp

```diff
@@ -51,6 +51,45 @@ TEST(CampResource, GetPlatform)
   ASSERT_EQ(static_cast<const Resource>(Omp()).get_platform(), Platform::omp_target);
 #endif
 }
+TEST(CampResource, Compare)
+{
+  Resource h1{Host()};
+  Resource h2{Host()};
+  Host h; Resource h3{h};
+
+#ifdef CAMP_HAVE_CUDA
+  Resource r1{Cuda()};
+  Resource r2{Cuda()};
+  Cuda s; Resource r3{s};
+#endif
+#ifdef CAMP_HAVE_HIP
+  Resource r1{Hip()};
+  Resource r2{Hip()};
+  Hip s; Resource r3{s};
+#endif
+#ifdef CAMP_HAVE_OMP_OFFLOAD
+  Resource r1{Omp()};
+  Resource r2{Omp()};
+  Omp s; Resource r3{s};
+#endif
+
+  ASSERT_TRUE(h1 == h1);
+  ASSERT_TRUE(h1 == h2);
+  ASSERT_TRUE(h3 == h);
```

Comment:
There should be coverage for all the cases between type-erased and typed resources.
```suggestion
  ASSERT_TRUE(h3 == h);
  ASSERT_FALSE(h3 != h);
  ASSERT_TRUE(h == h3);
  ASSERT_FALSE(h != h3);
```

Reply:
There should also be coverage for comparing h to h3, h to r3, s to h3 and s to r3.

Reply:
Ooo looks like I broke something... stay tuned lol

Reply:
Here's a way to wrap up all the comparisons for a pair of resources.
```suggestion
  auto assert_resources_equal = [](auto const& r1, auto const& r2) {
    ASSERT_TRUE(r1 == r2);
    ASSERT_TRUE(r2 == r1);
    ASSERT_FALSE(r1 != r2);
    ASSERT_FALSE(r2 != r1);
  };
  assert_resources_equal(h1, h2);
  assert_resources_equal(h3, h);
```